creusot-contracts/src/std/slice.rs

```diff
@@ -83,6 +83,96 @@ impl<T> SliceIndexSpec<[T]> for usize {
     }
 }
 
+impl<T> SliceIndexSpec<[T]> for Range<usize> {
+    #[predicate]
+    fn in_bounds(self, seq: Seq<T>) -> bool {
+        pearlite! { @self.start <= @self.end && @self.end <= seq.len() }
+    }
+
+    #[predicate]
+    fn has_value(self, seq: Seq<T>, out: [T]) -> bool {
+        pearlite! { seq.subsequence(@self.start, @self.end) == @out }
+    }
+
+    #[predicate]
+    fn resolve_elswhere(self, old: Seq<T>, fin: Seq<T>) -> bool {
+        pearlite! {
+            forall<i : Int> 0 <= i && (i < @self.start || @self.end <= i) && i < old.len()
```

Comment:
As a pragmatic issue I wonder if provers would prefer that this is specified using two seperate quantifiers, one for the lower segment and one for the upper segment?

I don't know the answer nor do I have a clear idea on the best way to test it..

Reply:
I have no idea what is more beneficial to provers, but once this PR is done we could run the test with both options, to see if one of them results in consistently fewer steps for the provers?